lib/Dialect/Triton/Transforms/Combine.cpp

```diff
@@ -101,6 +101,93 @@ class CombineSelectMaskedLoadPattern : public mlir::RewritePattern {
   }
 };
 
+// sum(x[:, :, None] * y[None, :, :], 1)
+// -> dot(x, y)
+class CombineBroadcastMulReducePattern : public mlir::RewritePattern {
+private:
+  static bool isAddF32(const Operation *op) {
+    if (auto addf = dyn_cast_or_null<arith::AddFOp>(op))
+      return addf.getType().getIntOrFloatBitWidth() <= 32;
```

Comment:
the name seems misleading as it would return true for addF16? Or is it meant to check `getType().isF32();`